device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -101,16 +100,8 @@
      */
     DeviceIO(DeviceClientConfig config, long sendPeriodInMilliseconds, long receivePeriodInMilliseconds)
     {
-        /* Codes_SRS_DEVICE_IO_21_002: [If the `config` is null, the constructor shall throw an IllegalArgumentException.] */
-        if(config == null)
```

Comment:
this check seems useful, why remove it?

Reply:
unless it happens in the public method that calls this one ...

Reply:
I can add this back in. The DeviceIO layer doesn't do much with the config object, but it does dereference it at least once so there is value in leaving this check.